### PR TITLE
New test - add command line and image regression tests for exiv2.

### PR DESCRIPTION
**Pull request by jwbruno**
Related ticket: https://progress.opensuse.org/issues/48149
Simple tests from exiv2 verifying metadata information and basic
functionalities.

Needles:
 https://github.com/os-autoinst/os-autoinst-needles-opensuse/pull/535

Verification runs:
- TW: http://10.161.229.244/tests/325#step/exiv2/1
- Leap: http://10.161.229.244/tests/323#step/exiv2/1

### CODE REVIEW

File: tests/x11/exiv2.pm [hunk 1]
```diff
@@ -0,0 +1,106 @@
+# SUSE's openQA tests
+#
+# Copyright © 2016-2019 SUSE LLC
```
Comment (czerw) [inline tests/x11/exiv2.pm]: should be only 2019

**Comment by czerw** [comment]
@jwbruno please properly rebase you pr, don't push changes which are not related and include/schedule test in main_common

File: lib/main_common.pm [hunk 2]
```diff
@@ -1447,6 +1447,7 @@ sub load_extra_tests_desktop {
         if (gnomestep_is_applicable()) {
             # Setup env for x11 regression tests
             loadtest "x11/x11_setup";
+            loadtest "x11/exiv2";
```
Comment (czerw) [inline lib/main_common.pm]: add condition that test is for opensuse only

File: tests/x11/exiv2.pm [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    assert_script_run "exiv2 -ep1 20190201_154421.jpg";
+    assert_script_run "ls -ld 20190201_154421-preview1.jpg";
+    script_run "eog 20190201_154421-preview1.jpg", 0;
+    assert_screen "exiv2_test_preview";
+    send_key 'alt-f4';
+
+    # clean-up
+    assert_script_run "rm 20190201_154421.jpg";
+    assert_script_run "rm 20190201_154421-preview1.jpg";
+    type_string "exit\n";
```
Comment (czerw) [inline tests/x11/exiv2.pm]: exit is not necessary

File: tests/x11/exiv2.pm [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+    exiv2_info_test;
+
+    assert_script_run "exiv2 rename test.jpg";
+    assert_script_run "ls -ld 20190201_154421.jpg";
+    script_run "eog 20190201_154421.jpg", 0;
+    assert_screen "exiv2_rename_test";
+    send_key 'alt-f4';
+
+    assert_script_run "exiv2 -ep1 20190201_154421.jpg";
```
Comment (czerw) [inline tests/x11/exiv2.pm]: come comments might be useful here to see what is it doing

File: tests/x11/exiv2.pm [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    become_root;
+    pkcon_quit;
+    zypper_call "in exiv2";
+    type_string "exit\n";
+
+    assert_script_run "wget --quiet " . data_url('exiv2/test.jpg') . " -O test.jpg";
+
+    exiv2_info_test;
+
+    assert_script_run "exiv2 rename test.jpg";
```
Comment (czerw) [inline tests/x11/exiv2.pm]: add comment about action

File: tests/x11/exiv2.pm (hunk 4 above)
Comment (jwbruno) [reply tests/x11/exiv2.pm]: ok, thank you for the suggestions

File: lib/main_common.pm (hunk 2 above)
Comment (jwbruno) [reply lib/main_common.pm]: This code is already under a "if (check_var('DISTRI', 'opensuse'))", should I really double check it on the load_test?

File: tests/x11/exiv2.pm (hunk 5 above)
Comment (jwbruno) [reply tests/x11/exiv2.pm]: I am not sure if the changes I did are ok

File: lib/main_common.pm (hunk 2 above)
Comment (czerw) [reply lib/main_common.pm]: ok, correct

**Comment by dzedro** [comment]
it will not run on SLE ?

**Comment by jwbruno** [comment]
> it will not run on SLE ?

No, actually the exiv2 application is not included on sle packages, only the libexiv2, so we changed the scope. One thing that could be done is to add a different test with sdk required and test some basic linkage on the library; but other than that (and that having to actually rebuilding exiv2 from src package, which was actually done on some regression tests) I see no way to test it on sle.

**Pull request merged**